src/cmake_generator.cpp

```diff
@@ -611,13 +611,28 @@ void generate_cmake(const char *path, const parser::Project *parent_project) {
         const auto &packages = project.vcpkg.packages;
         for (size_t i = 0; i < packages.size(); i++) {
             const auto &package = packages[i];
-            if (!vcpkg_valid_identifier(package)) {
-                throw std::runtime_error("Invalid vcpkg package name '" + package + "'");
+            if (!vcpkg_valid_identifier(package.name)) {
+                throw std::runtime_error("Invalid vcpkg package name '" + package.name + "'");
             }
-            ofs << "    \"" << package << '\"';
-            if (i + 1 < packages.size()) {
-                ofs << ',';
+            if (package.features.empty()) {
+                ofs << "    \"" << package.name << '\"';
+            } else {
+                ofs << "    {\n";
+                ofs << "      \"name\": \"" << package.name << "\",\n";
+                ofs << "      \"features\": [";
+                for (size_t j = 0; j < package.features.size(); j++) {
+                    const auto &feature = package.features[j];
```

Comment:
Do features also need a `vcpkg_valid_identifier` check?

Reply:
Added in b4b0f533a9d3b65ff0ac53c296d617c69f99c876